File: basicprogramming1/basicprogramming1.cpp
#include <bits/stdc++.h>

typedef long long ll;
using namespace std;

int main() {
    int n, t;
    cin >> n >> t;
    vector<int> array;
    for (int i = 0; i < n; i++) {
        int tmp;
        cin >> tmp;
        array.push_back(tmp);
    }

    switch (t) {
        case 1: {
            cout << "7\n";
            break;
        }
        case 2: {
            if (array[0] > array[1]) cout << "Bigger\n";
            else if (array[0] == array[1]) cout << "Equal\n";
            else cout << "Smaller\n";
            break;
        }
        case 3: {
            vector<int> newArray = {array[0], array[1], array[2]};
            sort(newArray.begin(), newArray.end());
            cout << newArray[1] << '\n';
            break;
        }
        case 4: {
            ll total = 0;
            for (auto i : array) {
                total+=i;
            }
            cout << total << '\n';
            break;
        }
        case 5: {
            ll total = 0;
            for (auto i : array) {
                total+=i % 2 ? 0 : i;
            }
            cout << total << '\n';
            break;
        }
        case 6: {
            for (auto i : array) {
                cout << (char) ((i % 26) + 97);
            }
            cout << endl;
            break;
        }
        case 7: {
            int i = array[0];
            array[0] = -1;
            // int cycles = 0;
            while (true) {
                if (i >= n) {
                    cout << "Out\n";
                    return 0;
                }
                if (i == n-1) {
                    cout << "Done\n";
                    return 0;
                }
                if (i == -1) {
                    cout << "Cyclic\n";
                    return 0;
                }
                int tmp = i;
                i = array[i];
                array[tmp] = -1;
            }
            cout << "Cyclic\n";
            // int i = 0;
            // int cycles = 0;
            // while (cycles < n + 2) {
            //     cycles++;
            //     i = array[i];
            //     if (0 > i || i >= n) {
            //         cout << "Out\n";
            //         return 0;
            //     } else if (i == n-1) {
            //         cout << "Done\n";
            //         return 0;
            //     }
            // }
            // cout << "Cyclic\n";
            break;
        }
        default:
            break;
    }
    
    return 0;
}

File: basicprogramming1/basicprogramming1.py
def main():
    n, t = list(map(int, input().split()))
    items = list(map(int, input().split()))

    if t == 1:
        print(7)
    elif t == 2:
        if items[0] == items[1]: print("Equal")
        else:
            print("Bigger" if items[0] > items[1] else "Smaller")
    elif t == 3:
        newItems = [items[0],items[1],items[2]]
        newItems.sort()
        print(newItems[1])

    elif t == 4:
        print(sum(items))
    elif t == 5:
        total = 0
        for item in items:
            total += 0 if item % 2 else item
        print(total)
    elif t == 6:
        mod26 = map(lambda x: x % 26, items)
        tochar = map(lambda x: chr(ord('a') + x), mod26)
        print(''.join(tochar))
    elif t == 7:
        index = items[0]
        items[0] = None
        while True:
            if index >= n:
                print('Out')
                break
            if index == n - 1:
                print('Done')
                break
            if items[index] == None:
                print('Cyclic')
                break
            items[index], index = None, items[index]
    return

if __name__ == "__main__":
    main()
